allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -217,29 +220,44 @@ def forward(self,  # type: ignore
 
         # Compute entity and question word similarity through a dot product.
-        question_table_similarity = torch.bmm(embedded_table.view(batch_size,
-                                                                  num_entities * num_entity_tokens,
-                                                                  self._embedding_dim),
-                                              torch.transpose(embedded_question, 1, 2))
+        question_entity_similarity = torch.bmm(embedded_table.view(batch_size,
+                                                                   num_entities * num_entity_tokens,
+                                                                   self._embedding_dim),
+                                               torch.transpose(embedded_question, 1, 2))
 
         # We divide the similarity scores by the embedding dim to reduce the variance of these
         # scores, and put the similarity scores into the same ballpark range as the linking
         # features and other scores in the model.  Glove vectors have average absolute magnitude on
         # the order of 1, which means that a 200-dimensional vector would have a dot product with
         # itself on the order of 200.  This is not reasonable to have as input to a softmax, which
         # we do later, so we need to scale by the number of dimensions.
-        question_table_similarity = question_table_similarity.view(batch_size,
-                                                                   num_entities,
-                                                                   num_entity_tokens,
-                                                                   num_question_tokens) / self._embedding_dim
+        question_entity_similarity = question_entity_similarity.view(batch_size,
+                                                                     num_entities,
+                                                                     num_entity_tokens,
+                                                                     num_question_tokens) / self._embedding_dim
         # (batch_size, num_entities, num_question_tokens)
-        question_table_similarity_max_score, _ = torch.max(question_table_similarity, 2)
+        question_entity_similarity_max_score, _ = torch.max(question_entity_similarity, 2)
+
         # (batch_size, num_entities, num_question_tokens, num_features)
         linking_features = table['linking']
-        linking_scores = question_table_similarity_max_score
+        linking_scores = question_entity_similarity_max_score
+
         if self._linking_params is not None:
             feature_scores = self._linking_params(linking_features).squeeze(3)
             linking_scores = linking_scores + feature_scores
+        else:
+            # pylint: disable=line-too-long
+            # The similarity between a question token and an entity's neighbors is useful when
```

Comment:
Can you add a note to the docstring for `num_linking_features` that says what we're doing with neighbor similarity when you give no features?  Also, the first sentence in the comment here should mention _what_ you're doing, because a reader doesn't know that.  Your second sentence can say _why_ you're doing this, which is what you currently have.

Reply:
done